### ISSUE TITLE
Convention check invalid-name 

### ISSUE THREAD

**Issue by aequitz** [PyCQA/pylint#7890]
### Bug description

Pylint provides an invalid-name convention message in an incorrect situation.

Code:
```python
some_list = ['Each', 'word', 'is', 'a', 'new', 'line.']
some_list_str = '\n'.join(some_list)
```

Pylint Error:
`"Constant name \"yaml_lines_str\" doesn't conform to UPPER_CASE naming style"`

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
[{
	"resource": "some_file.py",
	"owner": "_generated_diagnostic_collection_name_#1",
	"code": "C0103:invalid-name",
	"severity": 2,
	"message": "Constant name \"some_list_str\" doesn't conform to UPPER_CASE naming style",
	"source": "Pylint",
}]
```

### Expected behavior

I would expect that some_list_str is the appropriate snake_case naming style since it is not a constant. However, pylint seems to see the '\n' at the beginning of the statement and thinks it is a constant.

### Pylint version

```shell
pylint 2.15.5
astroid 2.12.12
Python 3.9.13 (tags/v3.9.13:6de2ca5, May 17 2022, 16:36:42) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

Windows 10

**Comment by mbyrnepr2** [comment]
Thanks for the report.
I think this is either invalid or a duplicate. Pylint considers strings defined at the module-level to be constants (i.e. must be UPPER_CASED). If you change the example to `some_list_str = ''.join(some_list)` the issue also appears.

As for being a duplicate, it has been discussed in various issues. https://github.com/PyCQA/pylint/issues/3585 is one but there may be a more accurate one somewhere.

I'll leave it open for now to invite some other thoughts on this.

**Comment by Pierre-Sassoulas** [comment]
Latest discussion about this was in https://github.com/PyCQA/pylint/pull/7808

**Comment by DanielNoord** [comment]
Closing this as there are other issues with "fuller" discussions.
